core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java

```diff
@@ -899,11 +901,82 @@ public boolean resolveConflictsOnMerge() {
+   }
+
+   @Override
+   public synchronized void queueConflictResolution(CacheTopology conflictTopology) {
+      if (resolveConflictsOnMerge()) {
+         log.debugf("Queueing conflict resolution for cache %s with members %s", cacheName, conflictTopology.getMembers());
+         queuedConflictTopology = conflictTopology;
+         startQueuedConflictResolution();
+      }
+   }
+
+   private synchronized void startQueuedConflictResolution() {
+      if (queuedConflictTopology == null)
+         return;
+
+      if (!conflictFuture.isDone()) {
+         // Check membership, if node removed then we must cancel the CR and start again below
+         // If currentTopology members size is one, cancel CR as it will be re-triggered on future merge
+         conflictFuture.cancel(true);
+      }
+
+      AdvancedCache<?, ?> cache = cacheManager.getCache(cacheName).getAdvancedCache();
+      InternalConflictManager<?, ?> conflictManager = (InternalConflictManager) ConflictManagerFactory.get(cache);
+      conflictFuture = conflictManager.resolveConflicts(queuedConflictTopology);
+      // Will execute using the StateTransferExecutor on completion
+      conflictFuture.thenRun(this::completeConflictResolution);
+   }
+
+   private synchronized void completeConflictResolution() {
+      CacheTopology conflictTopology = queuedConflictTopology;
+      queuedConflictTopology = null;
+      log.tracef("Cache %s conflict resolution complete", cacheName);
+      availabilityStrategy.onConflictResolutionEnd(this, conflictTopology);
+   }
+
+   @Override
+   public synchronized boolean restartConflictResolution(List<Address> members) {
+      // If queuedConflictTopology is null then no CR in progress
+      if (!resolveConflictsOnMerge() || queuedConflictTopology == null)
+         return false;
+
+      // No need to reattempt CR if only one node remains, so cancel CR, cleanup and queue rebalance
+      if (members.size() == 1) {
+         log.debugf("Cache %s cancelling conflict resolution as only one cluster member: members=%s", cacheName, members);
+         conflictFuture.cancel(true);
+         queuedConflictTopology = null;
+         return false;
+      }
+
+      // CR members are the same as newMembers, so no need to restart
+      if (members.containsAll(queuedConflictTopology.getMembers())) {
+         if (trace) log.tracef("Cache %s not restarting CR, existing conflict topology contains all new members (%s)", members);
+         return false;
+      }
+
+      CacheTopology t = queuedConflictTopology;
+      ConsistentHashFactory chf = getJoinInfo().getConsistentHashFactory();
+      ConsistentHash newHash = chf.updateMembers(t.getCurrentCH(), members, capacityFactors);
+
+      if (trace) log.tracef("Cache %s restarting conflict resolution with new consistent hash %s", cacheName, newHash);
+
+      queuedConflictTopology = new CacheTopology(t.getTopologyId() + 1, t.getRebalanceId() + 1,
```

Comment:
The topologyId needs to take into account the currentTopology Id, I'll push the change.